drivers/power/ltc4162l/ltc4162l.c

```diff
@@ -0,0 +1,829 @@
+
+	if (ret)
+		return ret;
+
+	*val = no_os_field_get(LTC4162L_CELL_COUNT_MASK, regval);
+
+	return 0;
+}
+
+/**
+ * @brief Get the chemistry type
+ * @param dev - The LTC4162L device structure.
+ * @param val - The chemistry type.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ltc4162l_get_chem_type(struct ltc4162l_desc *dev, uint16_t *val)
+{
+	int ret;
+	uint16_t regval;
+
+	ret = ltc4162l_reg_read(dev, LTC4162L_CHEM_CELLS_REG, &regval);
+	if (ret)
+		return ret;
+
+	*val = no_os_field_get(LTC4162L_CHEM_TYPE_MASK, regval);
+
+	return 0;
+}
+
+/**
+ * @brief Get the charger state
+ * @param dev - The LTC4162L device structure.
+ * @param val - The charger state.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ltc4162l_get_charger_state(struct ltc4162l_desc *dev, uint16_t *val)
+{
+	int ret;
+	uint16_t regval;
+
+	ret = ltc4162l_reg_read(dev, LTC4162L_CHARGE_STATUS, &regval);
+	if (ret)
+		return ret;
+
+	*val = regval;
+
+	return 0;
+}
+
+/**
+ * @brief Get the charge status
+ * @param dev - The LTC4162L device structure.
+ * @param val - The charge status.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ltc4162l_get_charge_status(struct ltc4162l_desc *dev, uint16_t *val)
+{
+	int ret;
+	uint16_t regval;
+
+	ret = ltc4162l_reg_read(dev, LTC4162L_CHARGE_STATUS, &regval);
+	if (ret)
+		return ret;
+
+	*val = regval;
+
+	return 0;
+}
+
+/**
+ * @brief Get the online status
+ * @param dev - The LTC4162L device structure.
+ * @param val - The online status.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ltc4162l_get_online_status(struct ltc4162l_desc *dev, uint16_t *val)
+{
+	int ret;
+	uint16_t regval;
+
+	ret = ltc4162l_reg_read(dev, LTC4162L_SYSTEM_STATUS_REG, &regval);
+	if (ret)
+		return ret;
+
+	*val = no_os_field_get(LTC4162L_ONLINE_STATUS_BIT, regval);
+
+	return 0;
+}
+
+/**
+ * @brief Get the battery voltage
+ * @param dev - The LTC4162L device structure.
+ * @param val - The battery voltage.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ltc4162l_read_vbat(struct ltc4162l_desc *dev, uint16_t *val)
+{
+	uint16_t regval;
+	int ret;
+	uint16_t cell_count;
+	uint16_t chem_type;
+
+	ret = ltc4162l_reg_read(dev, LTC4162L_VBAT, &regval);
+	if (ret)
+		return ret;
+
+	ret = ltc4162l_get_cell_count(dev, &cell_count);
+	if (ret)
+		return ret;
+
+	ret = ltc4162l_get_chem_type(dev, &chem_type);
+	if (ret)
+		return ret;
+
+	switch (dev->type) {
+	case ID_LTC4162L:
+	case ID_LTC4162F:
+		regval *= 1924;
+		regval *= cell_count;
+		regval /= 10;
+		*val = regval;
+		return 0;
+	case ID_LTC4162S:
+		regval *= 3848;
+		regval *= cell_count / 2;
+		regval /= 10;
+		*val = regval;
+		return 0;
+	case ID_LTC4015:
+		regval *= chem_type <= 6 ? 192264 : 128176;
+		regval *= cell_count;
+		regval /= 1000;
+		*val = regval;
+		return 0;
+	default:
+		return -EINVAL;
+	}
```

Comment:
Add some macros for these numbers (and maybe a comment describing what are the units).

Reply:
I have created new fcompute unctions to support chip_info properties and added some description for the units